docs/language-service.md

```diff
@@ -0,0 +1,38 @@
+# Key features of the Bicep language service
```

Comment:
We also have the following currently:
- diagnostics (errors/warnings/squiggles) - not sure if worth mentioning
- document symbols - this powers the Outline view, go to symbol commands as well as the navigation breadcrumbs at the top of the file
- highlights - when you click on a symbol, we immediately highlight all references to that symbol in the file (double click on any word highlights all occurrences of the word via text search)
- hovers

Just for reference, the things in the LSP spec that we have NOT implemented are:
- go to declaration (not sure if applicable)
- go to implementation (not applicable)
- code lens (need to experiment)
- signature help (planned for 0.3)
- range formatting/type formatting (auto formatting while typing) - @shenglol both could be very interesting for us, but especially the 2nd one
- folding range (low pri - should do eventually)
- selection range (i'm guessing it helps with text selection when you double click - we should do that for 0.3+)